akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -0,0 +1,234 @@
+   * containing the `ActorRef` of the current child, if any.
+   *
+   * The `BackoffSupervisor`delegates all messages from the child to the parent of the
+   * `BackoffSupervisor`, with the supervisor as sender.
+   *
+   * The `BackoffSupervisor` forwards all other messages to the child, if it is currently running.
+   *
+   * The child can stop itself and send a [[akka.actor.PoisonPill]] to the parent supervisor
+   * if it wants to do an intentional stop.
+   *
+   * Exceptions in the child are handled with the default supervisionStrategy, which can be changed by using
+   * [[BackoffOptions#withSupervisorStrategy]] or [[BackoffOptions#withDefaultStoppingStrategy]]. A
+   * `Restart` will perform a normal immediate restart of the child. A `Stop` will
+   * stop the child, but it will be started again after the back-off duration.
+   *
+   * @param childProps the [[akka.actor.Props]] of the child actor that
+   *   will be started and supervised
+   * @param childName name of the child actor
+   * @param minBackoff minimum (initial) duration until the child actor will
+   *   started again, if it is terminated
+   * @param maxBackoff the exponential back-off is capped to this duration
+   * @param randomFactor after calculation of the exponential back-off an additional
+   *   random delay based on this factor is added, e.g. `0.2` adds up to `20%` delay.
+   *   In order to skip this additional delay pass in `0`.
+   */
+  def onStop(
+    childProps: Props,
+    childName: String,
+    minBackoff: FiniteDuration,
+    maxBackoff: FiniteDuration,
+    randomFactor: Double): BackoffOptions =
+    BackoffOptionsImpl(StopImpliesFailure, childProps, childName, minBackoff, maxBackoff, randomFactor)
+}
+
+/**
+ * Configures a back-off supervisor actor. Start with `BackoffOptions.onStop` or `BackoffOptions.onFailure`.
+ * BackoffOptions is immutable, so be sure to chain methods like:
+ * {{{
+ * val options = BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ *               .withManualReset
+ * context.actorOf(BackoffSupervisor.props(options), name)
+ * }}}
+ * or capture the result of a method in a variable:
+ * {{{
+ * var options = BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ * options = options.withManualReset
+ * context.actorOf(BackoffSupervisor.props(options), name)
+ * }}}
+ */
+trait BackoffOptions {
+  /**
+   * Returns a new BackoffOptions with automatic back-off reset.
+   * The back-off algorithm is reset if the child does not crash witin the specified `resetBackoff`.
+   * @param resetBackoff The back-off is reset if the child does not crash within this duration.
+   */
+  def withAutoReset(resetBackoff: FiniteDuration): BackoffOptions
+
+  /**
+   * Returns a new BackoffOptions with manual back-off reset. The back-off is only reset
+   * if the child sends a `BackoffSupervisor.Reset` to its parent (the backoff-supervisor actor).
```

Comment:
hah, just noticed that `[[]] ()` above was rendered as markdown by github in above comment ;-)

Hm, I see `[[]]` may be troublesome because inner object; Ignore this comment then, not a big issue anyway